quickwit/quickwit-actors/src/mailbox.rs

```diff
@@ -320,4 +381,115 @@ mod tests {
+            }
+            Ok(())
+        }
+    }
+
+    #[tokio::test]
+    async fn test_mailbox_send_with_backpressure_counter_no_backpressure_cleansheet() {
+        let universe = Universe::new();
+        let back_pressure_actor = BackPressureActor;
+        let (mailbox, _handle) = universe.spawn_builder().spawn(back_pressure_actor);
+        // We send a first message to make sure the actor has been properly spawned and is listening
+        // for new messages.
+        mailbox
+            .ask_with_backpressure_counter(Duration::default(), None)
+            .await
+            .unwrap();
+        // At this point the actor was started and even processed a message entirely.
+        let backpressure_microsecs_counter =
+            IntCounter::new("test_counter", "help for test_counter").unwrap();
+        let wait_duration = Duration::from_millis(1);
+        let processed = mailbox
+            .send_message_with_backpressure_counter(
+                wait_duration,
+                Some(&backpressure_microsecs_counter),
+            )
+            .await
+            .unwrap();
+        assert_eq!(backpressure_microsecs_counter.get(), 0u64);
+        processed.await.unwrap();
+        assert_eq!(backpressure_microsecs_counter.get(), 0u64);
+    }
+
+    #[tokio::test]
+    async fn test_mailbox_send_with_backpressure_counter_backpressure() {
+        let universe = Universe::new();
+        let back_pressure_actor = BackPressureActor;
+        let (mailbox, _handle) = universe.spawn_builder().spawn(back_pressure_actor);
+        // We send a first message to make sure the actor has been properly spawned and is listening
+        // for new messages.
+        mailbox
+            .ask_with_backpressure_counter(Duration::default(), None)
+            .await
+            .unwrap();
+        let backpressure_microsecs_counter =
+            IntCounter::new("test_counter", "help for test_counter").unwrap();
+        let wait_duration = Duration::from_millis(1);
+        mailbox
+            .send_message_with_backpressure_counter(
+                wait_duration,
+                Some(&backpressure_microsecs_counter),
+            )
+            .await
+            .unwrap();
+        // That second message will present some backpressure, since the capacity is 0 and
+        // the first message willl take 1000 microsecs to be processed.
+        mailbox
+            .send_message_with_backpressure_counter(
+                Duration::default(),
+                Some(&backpressure_microsecs_counter),
+            )
+            .await
+            .unwrap();
+        assert!(backpressure_microsecs_counter.get() > 1_000u64);
+    }
+
+    #[tokio::test]
+    async fn test_mailbox_waiting_for_processing_does_not_counter_as_backpressure() {
+        let universe = Universe::new();
+        let back_pressure_actor = BackPressureActor;
+        let (mailbox, _handle) = universe.spawn_builder().spawn(back_pressure_actor);
+        mailbox
```

Comment:
question: why do you need to send do `.ask_with_backpressure_counter(Duration::default(), None)` before executing the real test?